src/Raven.Client/ServerWide/Operations/CompareExchange.cs

```diff
@@ -65,44 +107,13 @@ public override HttpRequestMessage CreateRequest(JsonOperationContext ctx, Serve
 
             public override void SetResponse(BlittableJsonReaderObject response, bool fromCache)
             {
-
-                if (response.TryGet(nameof(RawClusterValueResult.Index), out long index) == false)
-                    ThrowInvalidResponse();
-
-                response.TryGet(nameof(RawClusterValueResult.Value), out BlittableJsonReaderObject raw);
-
-                T result;
-                object val = null;
-                raw?.TryGet("Object", out val);
-
-                if (val == null)
-                {
-                    Result = new ClusterValueResult<T>
-                    {
-                        Index = index,
-                        Value = default(T)
-                    };
-                    return;
-                }
-                if (val is BlittableJsonReaderObject obj)
-                {
-                    result = (T)EntityToBlittable.ConvertToEntity(typeof(T), "cluster-value", obj, _conventions);
-                }
-                else
-                {
-                    raw.TryGet("Object", out result);
-                }
-
-                Result = new ClusterValueResult<T>
-                {
-                    Index = index,
-                    Value = result
-                };
+                Result = CmpXchgResult<T>.ParseFromBlittable(response, _conventions);
             }
         }
+
     }
 
-    public class CompareExchangeOperation<T> : IServerOperation
+    public class CompareExchangeOperation<T> : IOperation<CmpXchgResult<T>>
```

Comment:
I think we should name it better so it would better describe what it does. PutCompareExchangeOperation? CreateCompareExchangeOperation? Not sure.

Reply:
I need concrete names please :)
I would prefer simply CompareExchange without the 'Operation' ending (I know it is the convention, but in my opinion not a user friendly one)